### ISSUE TITLE
Don't Require Labels

### ISSUE THREAD

**Issue by cancan101** [NVIDIA/DIGITS#109]
Why is a labels file even required?

Rather than requiring the user to have the set files look like `"path int_label"` and then requiring a labels file that maps the class int to class name, allow for `"path string_label"` and then use something like http://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.LabelEncoder.html so covert the strings to ints.

![image](https://cloud.githubusercontent.com/assets/51059/7637218/43d5bcb4-fa3a-11e4-96c6-8965be26f77c.png)

**Comment by lukeyeager** [comment]
> Why is a labels file even required?

To answer your question, there are a couple of reasons for why DIGITS does this at the moment:

1. That's the format that caffe uses, so that's what I started with.
2. Having a labels file lets you use long label names with spaces and punctuation which would be a bit tricky to parse in a text file, especially when you can use a URL or a path for the image.

But this is a good suggestion. It might make textfile generation a bit simpler for people, and it would definitely make it easier to analyze the textfiles with the naked eye.
```
/path/to/image1.jpg cat
/path/to/image2.jpg dog
```
instead of 
```
/path/to/image1.jpg 0
/path/to/image2.jpg 1
```

I can't make this a high priority right now, but PR's are welcome!
